src/CP/core/model.jl

```diff
@@ -44,7 +44,7 @@ mutable struct Statistics
     objectiveUpPruning                      ::Union{Nothing, Float32}
     lastVar                                 ::Union{Nothing, AbstractIntVar} #last var on which we branched
     lastVal                                 ::Union{Nothing, Int} #last val on which we branched
-    searchTreeSize                          ::Union{Nothing, Int} #size of the search tree for impact based search
+    searchTreeSize                          ::Union{Nothing, StateObject{Int}} #size of the search tree for impact based search

```

Comment:
`searchTreeSize` is now a `StateObject{Int}`. This makes the variable dependent of the model.trailer that ensure that the value is updated once the search backtrack in the searchTree.

Reply:
Therefore, `searchTreeSize` has now to be updated using the function `setValue!`